src/Microsoft.ML.PCA/PcaTrainer.cs

```diff
@@ -347,14 +347,14 @@ protected override AnomalyPredictionTransformer<PcaModelParameters> MakeTransfor
             Desc = "Train an PCA Anomaly model.",
             UserName = UserNameValue,
             ShortName = ShortName)]
-        internal static CommonOutputs.AnomalyDetectionOutput TrainPcaAnomaly(IHostEnvironment env, Arguments input)
+        internal static CommonOutputs.AnomalyDetectionOutput TrainPcaAnomaly(IHostEnvironment env, Options input)
```

Comment:
Same as above; these can be kept `public` and the whole class can be made `internal`/`BestFriend`.

Reply:
The `Options` class above should be public.  Hence we cannot make entire class internal.

---
In reply to: [254391534](https://github.com/dotnet/machinelearning/pull/2390#discussion_r254391534) [](ancestors = 254391534)